### ISSUE TITLE
Jetson TK1: Shared Memory

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/picongpu#631]
New boards like the Jetson TK1 share the RAM between GPU and CPU (ARM).

  "The Jetson board has 2GB of physical memory that is shared by the ARM
CPU and the CUDA GPU." [arrayfire](http://arrayfire.com/zero-copy-on-tegra-k1/)

Due to double buffering, the "available" [gpu memory in `MySimulation.hpp`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/f0051b7f1c33a716fcee8cf91657f78b96e0bffd/src/picongpu/include/simulationControl/MySimulation.hpp#L274-L275) should be split by `2` (work-around by @psychocoderHPC):

```c++
freeGpuMem /= 2;
```

Also, they are `sm_32` (some of our examples already require `sm_35`).

**Comment by ax3l** [comment]
fun fact: the two dev kits seem to be [the only models](https://developer.nvidia.com/cuda-gpus) with `sm_32` right now.

**Comment by ax3l** [comment]
this might be an efficient way to find out if we are on a SoC/shared memory device:

```c++
#include <cuda.h>
#include <iostream> // cout
#include <cstring>  // memset

int main()
{
    size_t freeInternal = 0;
    size_t totalInternal = 0;

    cudaMemGetInfo(&freeInternal, &totalInternal);
    std::cout << freeInternal/1024/1024 << std::endl;

    char* i = new char[freeInternal];
    memset(i,'-',freeInternal);
//    cudaDeviceSynchronize();

    cudaMemGetInfo(&freeInternal, &totalInternal);
    std::cout << freeInternal/1024/1024 << std::endl;
    delete [] i;

    cudaMemGetInfo(&freeInternal, &totalInternal);
    std::cout << freeInternal/1024/1024 << std::endl;

    return 0;
}
```

**Comment by ax3l** [comment]
closed with #633
